Source/CompanyCommunicator.Common/Configuration/ConfigurationFactory.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Microsoft.Teams.Apps.CompanyCommunicator.Common.Configuration
+{
+    /// <summary>
+    /// Configuration factory returns relevant configuration for a given environment.
+    /// </summary>
+    public class ConfigurationFactory
+    {
+        private readonly string tenantId;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConfigurationFactory"/> class.
+        /// </summary>
+        /// <param name="tenantId">Tenant id.</param>
+        public ConfigurationFactory(string tenantId)
+        {
+            this.tenantId = tenantId ?? throw new System.ArgumentNullException(nameof(tenantId));
+        }
+
+        /// <summary>
+        /// Configuration factory returns relevant configuration for a given environment.
+        /// </summary>
+        /// <param name="env">Teams environment.</param>
+        /// <returns>App configurstion.</returns>
+        public IAppConfiguration GetAppConfiguration(string env)
+        {
+            switch (env)
+            {
+                case "GCCH":
```

Comment:
Should this be in an enum to ensure that this covers all environments?

Reply:
We could use an enum. Would need an additional change to convert string to enum when reading config.